src/sql/workbench/browser/modelComponents/iconUtils.ts

```diff
@@ -52,3 +52,7 @@ function getIconUri(iconPath: string | URI): URI {
 		return URI.revive(iconPath);
 	}
 }
+
+export function getIconKey(iconPath: IUserFriendlyIcon): string {
+	return getLightIconUri(iconPath).toString(true) + getDarkIconUri(iconPath)?.toString(true);
```

Comment:
Why are you using the path for generating the key? The extension already defines a key value for these so why wouldn't you just use that?

Reply:
Key value - you mean ariaLabel? User could pass no-unique value, so it's safer to generate a key internally.

Reply:
Ok I thought you were still having the user pass in the initial map when defining the columns. Why didn't you keep that? If you're going to have a map having it be done that way makes more sense to me.

But anyways this seems fine I guess.

Reply:
Users just passing iconPath objects at data row. This map should be actually moved into iconUtil module as it was discussed previously. createIconCssClass shouldn't create the css class for the same icon every time it been called. 
This change could be done along with a feature to allow ADS users to use embedded codicon icon set icons.